src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -405,15 +443,15 @@ internal static bool IsMsOrElastic(byte[] array)
 
 			for (var i = 0; i < 8; i++)
 			{
-				if (array[i] != elasticToken[i])
+				if (isElasticApm && array[i] != elasticToken[i])
```

Comment:
Are these changes with `and`-ing the array item with the `bool` variable meant to be an optimization?

Reply:
it was a kind of. After deeper analysis with benchmarks it doesn't make sense: differences within the limit of error

Reply:
I see. Yeah, I agree with your decision.

Also huge 👍 for measuring it!